wolfcrypt/src/port/Espressif/esp_crt_bundle/esp_crt_bundle.c

```diff
@@ -0,0 +1,1380 @@
+ * https://components.espressif.com/components/wolfssl/wolfssl
+ */
+#if defined(WOLFSSL_ESPIDF_COMPONENT_VERSION)
+    #if (WOLFSSL_ESPIDF_COMPONENT_VERSION > 0)
+        #define WOLFSSL_ESPIDF_COMPONENT_VERSION_VALID 1
+    #else
+        #define WOLFSSL_ESPIDF_COMPONENT_VERSION_VALID 0
+        #warning "This library depends on a recent version of wolfSSL config"
+    #endif
+#else
+    #warning "This library depends on a recent version of wolfSSL config"
+    #define WOLFSSL_ESPIDF_COMPONENT_VERSION_VALID -1
+#endif
+
+#include <wolfssl/wolfcrypt/port/Espressif/esp_crt_bundle.h>
+
+/* Bundle debug may come from user_settings.h and/or sdkconfig.h */
+#if defined(CONFIG_WOLFSSL_DEBUG_CERT_BUNDLE) || \
+    defined(       WOLFSSL_DEBUG_CERT_BUNDLE)
+    /* We'll only locally check this one: */
+    #undef         WOLFSSL_DEBUG_CERT_BUNDLE
+    #define        WOLFSSL_DEBUG_CERT_BUNDLE
+    /* Only display certificate bundle debugging messages when enabled: */
+    #define ESP_LOGCBI ESP_LOGI
+    #define ESP_LOGCBW ESP_LOGW
+    #define ESP_LOGCBV ESP_LOGV
+#else
+    /* Only display certificate bundle messages for most verbose setting.
+     * Note that the delays will likely cause TLS connection failures. */
+    #define ESP_LOGCBI ESP_LOGV
+    #define ESP_LOGCBW ESP_LOGV
+    #define ESP_LOGCBV ESP_LOGV
+#endif
+
+#if defined(WOLFSSL_EXAMPLE_VERBOSITY)
+    #define ESP_LOGXI ESP_LOGI
+    #define ESP_LOGXW ESP_LOGW
+    #define ESP_LOGXV ESP_LOGW
+#else
+    #define ESP_LOGXI ESP_LOGV
+    #define ESP_LOGXI ESP_LOGV
+    #define ESP_LOGXI ESP_LOGV
+#endif
+
+#ifndef X509_MAX_SUBJECT_LEN
+    #define X509_MAX_SUBJECT_LEN 255
+#endif
+
+#define IS_WOLFSSL_CERT_BUNDLE_FORMAT
+#ifndef IS_WOLFSSL_CERT_BUNDLE_FORMAT
+    /* For reference only, the other cert bundles are structured differently!
+     * The others contain only a PARTIAL certificate, along with a name. */
+    #define BUNDLE_HEADER_OFFSET 2
+    #define CRT_HEADER_OFFSET 4
+#else
+    /* Note these are also set in [ESP-IDF]/components/esp-tls/esp_tls_wolfssl.c
+     * to avoid conflicts with other cert bundles that may, in theory,
+     * be enabled concurrently (NOT recommeneded).
+     *
+     * Ensure they exactly match here: */
+    #define BUNDLE_HEADER_OFFSET 2
+    #define CRT_HEADER_OFFSET 2
+#endif
+
+/* NOTE: Manually edit sort order in gen_crt_bundle.py
+ *
+ * The default is having the bundle pre-sorted in the python script
+ * to allow for rapid binary cert match search at runtime. The unsorted
+ * seach ALWAYS works, but when expecting a sorted search the python
+ * script MUST presort the data, oherwise the connection will likely fail.
+ *
+ * When debugging and using an unsorted bundle, define CERT_BUNDLE_UNSORTED
+ * Reminder: the actual sort occurs in gen_crt_bundly.py call from CMake. */
+/* #define CERT_BUNDLE_UNSORTED */
+
+
+/* Inline cert bundle functions performance hint unless otherwise specified. */
+#ifndef CB_INLINE
```

Comment:
Why provide an option to inline these?

Reply:
Good question. Some of the Cert Bundle code is called _during an active TLS connection handshake_, from a callback. This `inline` allows the end user to squeak out every ounce of performance (or not) as desired.